src/libcore/iter/traits.rs

```diff
@@ -473,15 +473,28 @@ impl<'a, I: DoubleEndedIterator + ?Sized> DoubleEndedIterator for &'a mut I {
 /// impl ExactSizeIterator for Counter {
 ///     // We already have the number of iterations, so we can use it directly.
 ///     fn len(&self) -> usize {
-///         self.count
+///         6 - self.count
 ///     }
 /// }
 ///
 /// // And now we can use it!
 ///
 /// let counter = Counter::new();
```

Comment:
needs to be `mut`. I would use just two assertions here, we don't need them all (it's supposed to be example, instructive code, not an exhaustive test).